### PR TITLE
Sort Workflow instances

### LINKED ISSUE #5376: Workflow instances: Allow to sort
Workflow instances should be displayed by creation date descending and also be allowed to be sorted by date ascending.

### PR DESCRIPTION
**Pull request by agriffard**
Fixes #5376

### CODE REVIEW

**Comment by agriffard** [comment]
image

File: src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowController.cs [hunk 1]
```diff
@@ -101,6 +101,19 @@ public async Task<IActionResult> Index(int workflowTypeId, WorkflowIndexViewMode
                     break;
             }
 
+            switch (model.Options.OrderBy)
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowController.cs]:
```csharp
var query = model.Options.OrderBy switch
{
    WorkflowOrder.CreatedDesc => query.OrderByDescending(x => x.CreatedUtc),
    WorkflowOrder.Created => query.OrderBy(x => x.CreatedUtc),
    _ =>query.OrderByDescending(x => x.CreatedUtc)
};
```

File: src/OrchardCore.Modules/OrchardCore.Workflows/ViewModels/WorkflowIndexViewModel.cs [hunk 2]
```diff
@@ -27,6 +27,11 @@ public WorkflowIndexOptions()
 
         public WorkflowBulkAction BulkAction { get; set; }
         public WorkflowFilter Filter { get; set; }
+
+        public WorkflowOrder OrderBy { get; set; }
+        [BindNever]
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Workflows/ViewModels/WorkflowIndexViewModel.cs]: Add new line before

File: src/OrchardCore.Modules/OrchardCore.Workflows/ViewModels/WorkflowIndexViewModel.cs [hunk 3]
```diff
@@ -27,6 +27,11 @@ public WorkflowIndexOptions()
...
+        public List<SelectListItem> WorkflowsSorts { get; set; }
+
         [BindNever]
         public List<SelectListItem> WorkflowsStatuses { get; set; }
         [BindNever]
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Workflows/ViewModels/WorkflowIndexViewModel.cs]: Also here

**Pull request merged**